File: app/src/main/java/aipova/githubsearch/dao/GitHubApi.java
package aipova.githubsearch.dao;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

/**
 * Created by aipova on 09.02.17.
 */

public interface GitHubApi {
    public static final String GITHUB_API_URL = "https://api.github.com/";

    @GET("/search/repositories")
    Call<RepoSearchResult> getRepositories(@Query("q") String q, @Query("sort") String sort,
                                           @Query("order") String order, @Query("page") String page,
                                           @Query("per_page") String perPage);
}

File: app/src/main/java/aipova/githubsearch/dao/GitHubApi.kt
package aipova.githubsearch.dao

import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Query

/**
 * Created by aipova on 09.02.17.
 */

interface GitHubApi {

    @GET("/search/repositories")
    fun getRepositories(
        @Query("q") q: String, @Query("sort") sort: String,
        @Query("order") order: String, @Query("page") page: String,
        @Query("per_page") perPage: String
    ): Call<RepoSearchResult>

    companion object {
        val GITHUB_API_URL = "https://api.github.com/"
    }
}
